tensorflow/python/platform/self_check.py

```diff
@@ -74,12 +75,22 @@ def preload_check():
         try:
           ctypes.WinDLL(build_info.cudart_dll_name)
         except OSError:
+          with open("cuda_links.json") as f:
+            link_dict = json.load(f)
           raise ImportError(
               "Could not find %r. TensorFlow requires that this DLL be "
               "installed in a directory that is named in your %%PATH%% "
               "environment variable. Download and install CUDA %s from "
-              "this URL: https://developer.nvidia.com/cuda-90-download-archive"
-              % (build_info.cudart_dll_name, build_info.cuda_version_number))
+              "this URL: %s"
```

Comment:
Why not just use https://developer.nvidia.com/cuda-downloads ?
You can click "legacy releases" and get the other links.

Reply:
Two reasons:
1. It would save them a click if we can directly link them to the required version download.
2. For version which have multiple updates, new users might get confused about which one to prefer. Fixing it to the latest patch would clear the ambiguity.